music_assistant/controllers/media/playlists.py

```diff
@@ -44,6 +45,13 @@ def __init__(self, *args, **kwargs) -> None:
             "music/playlists/remove_playlist_tracks", self.remove_playlist_tracks
         )
 
+    def _verify_update_allowed(self, update: Playlist) -> None:
+        # Additional validation for non-streaming providers to prevent path traversal
+        for update_mapping in update.provider_mappings:
+            provider = self.mass.get_provider(update_mapping.provider_instance)
+            if provider and not provider.is_streaming_provider:
+                validate_playlist_filename(update_mapping.item_id)
+
```

Comment:
adjust this function to receive the old provider mappings and the new provider mappings.
then work out which provider mappings belong to a filesystem provider and simply deny changing the item _id for the same instance_id - that works because in case of non-streaming providers the playlist item will be unique